### PR TITLE
Featuredata react conversion select features by properties

### PR DESCRIPTION
**Pull request by DenverCoder544**

### CODE REVIEW

File: bundles/framework/featuredata/view/SelectByProperties.jsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+const StyledButton = styled(Button)`
+    margin: 0 .5em 0 .5em
+`;
+
+const Buttons = (props) => {
+    const { closePopup, removeFilter, applyFilters, hasErrors } = props;
+    return <ButtonsContainer>
+        <StyledButton onClick={() => { closePopup(); }}>
+            <Message bundleKey={FEATUREDATA_BUNDLE_ID} messageKey={'selectByPropertiesPopup.buttons.cancel'}/>
+        </StyledButton>
```
Comment (ZakarFin) [inline bundles/framework/featuredata/view/SelectByProperties.jsx:91]: Try replacing this with https://github.com/oskariorg/oskari-frontend/blob/develop/src/react/components/buttons/SecondaryButton.jsx with type `cancel` so we don't need to have the localization in multiple places.

Maybe we could even replace the clearAll with `<SecondaryButton type='reset' />`?

File: bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js [hunk 2]
```diff
@@ -152,12 +159,28 @@ class FeatureDataPluginUIHandler extends StateHandler {
...
 
+    openSelectByPropertiesPopup () {
+        if (this.popupController) {
+            this.closePopup();
+            return;
+        }
+        this.popupController = showSelectByPropertiesPopup(this.getState(), this.getController());
+    }
+
+    closePopup () {
```
Comment (ZakarFin) [inline bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js:177]: I'm not sure if there will be other popups here, but we should rename this to match the opener-function/something like closeSelectByPropertiesPopup() to make it immediately clear what popups it is closing.

File: bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js [hunk 3]
```diff
@@ -170,18 +193,148 @@ class FeatureDataPluginUIHandler extends StateHandler {
...
         if (this.flyoutController) {
             this.flyoutController.close();
             this.flyoutController = null;
         }
-        this.updateState({ flyoutOpen: false, layers: resetLayers ? null : this.getState().layers });
+
+        if (this.popupController) {
+            this.popupController.close();
+            this.popupController = null;
+        }
```
Comment (ZakarFin) [inline bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js:214]:
```suggestion
       this.closePopup();
```
Or the renamed version.

File: bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js [hunk 4]
```diff
@@ -170,18 +193,148 @@ class FeatureDataPluginUIHandler extends StateHandler {
...
+                filteredList = filterFeaturesByAttribute(filteredList, filter);
+            });
+            filteredList
+                .map((feature) => feature.id)
+                .filter(id => !!id)
+                .forEach(id => filteredIds.add(id));
+        });
+
+        this.selectionService.setSelectedFeatureIds(activeLayerId, [...filteredIds]);
+        this.updateState({ selection: [...filteredIds] });
```
Comment (ZakarFin) [inline bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js:286]: We should remove updateState() from here and do it when we receive the event about selected features. That way we don't need to care how the features got selected and selecting from map works as well as selecting through this tool.

File: bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js (hunk 2 above)
Comment (DenverCoder544) [reply bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js]: This was on my todo list as an item to be forgotten. Fixed.

File: bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js (hunk 4 above)
Comment (DenverCoder544) [reply bundles/framework/featuredata/plugin/FeatureDataPluginHandler.js]: Removed excessive state update.

File: bundles/framework/featuredata/view/SelectByProperties.jsx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+                <StyledSelectSmall
+                    options={generateLogicalOperatorOptions()}
+                    value={filter.logicalOperator}
+                    onChange={(value) => { filter.logicalOperator = value; updateFilters(index, filter); }}
+                />
+            }
+            { showAddRemove &&
+                <>
+                    <Button onClick={(() => addFilter())}><PlusCircleTwoTone/></Button>
+                    { showRemove && <Button onClick={(() => removeFilter(index))}><MinusCircleTwoTone/></Button> }
```
[the remark this answers is not in the document]
Comment (DenverCoder544) [reply bundles/framework/featuredata/view/SelectByProperties.jsx]: changed the buttons to the look of those in myplaces. IMO some kind of confirmation dialog would be more annoying than helpful, after all we're not resetting anything _that_ significant here

**Pull request merged**